BurnOutSharp/PackerType/WinZipSFX.cs

```diff
@@ -0,0 +1,494 @@
+        {
+            /* If false positives do end up being an issue, there's one more consistnet string that may be used:
+
+               "_winzip_" (Found in almost every WinZip SFX archive, and every single 3+ one.)
+               Hex: 0x5F, 0x77, 0x69, 0x6E, 0x7A, 0x69, 0x70, 0x5F */
+
+            // Find the end of the XML so it can be imported and parsed
+            // </assembly>
+            byte[] check = new byte[] { 0x3C, 0x2F, 0x61, 0x73, 0x73, 0x65, 0x6D, 0x62, 0x6C, 0x79, 0x3E };
+            if (fileContent.Contains(check, out int position2))
+            {
+                int offset = position2 + 11 - xmlStartPosition;
+                string XML = Encoding.ASCII.GetString(fileContent, xmlStartPosition, offset);
+
+                // Load and parse the XML to find the version string, and run the Version 2.X finder if the XML fails to parse.
+                try
+                {
+                    XmlDocument xmlDoc = new XmlDocument();
+                    xmlDoc.LoadXml(XML);
+                    string xmlVersion = xmlDoc["assembly"]["assemblyIdentity"].GetAttributeNode("version").InnerXml;
+
+                    // 3.0.7158 is the only 3+ version that doesn't contain the "W.i.n.Z.i.p. .S.e.l.f.-.E.x.t.r.a.c.t.o.r." string.
+                    if (xmlVersion == "3.0.7158.0")
+                        return "3.0.7158";
+
+                    // "W.i.n.Z.i.p. .S.e.l.f.-.E.x.t.r.a.c.t.o.r."
+                    check = new byte[] { 0x57, 0x00, 0x69, 0x00, 0x6E, 0x00, 0x5A, 0x00, 0x69, 0x00, 0x70, 0x00, 0x20, 0x00, 0x53, 0x00, 0x65, 0x00, 0x6C, 0x00, 0x66, 0x00, 0x2D, 0x00, 0x45, 0x00, 0x78, 0x00, 0x74, 0x00, 0x72, 0x00, 0x61, 0x00, 0x63, 0x00,0x74, 0x00, 0x6F, 0x00, 0x72, 0x00 };
+                    if (fileContent.Contains(check, out int position3))
+                    {
+                        // Some version strings don't exactly match the public version number
+                        switch (xmlVersion)
+                        {
+                            case "3.1.7556.0":
+                                return "3.1.7556";
+                            case "3.1.8421.0":
+                                return "4.0.8421";
+                            case "3.1.8672.0":
+                                return "4.0.8672";
+                            case "4.0.1221.0":
+                                return "4.0.12218";
+                            default:
+                                return $"(Unknown Version - {xmlVersion})";
+                        }
+                    }
+                    return GetV2Version(fileContent, position);
+                }
+                catch
+                {
+                    return GetV2Version(fileContent, position);
+                }
+            }
+            return GetV2Version(fileContent, position);
+        }
+
+        private static string GetV2Version(byte[] fileContent, int position)
```

Comment:
As voiced on Discord, I *really* don't like this series of checks to get the version, but until we find a way to get the version string directly, this will suffice.